### PR TITLE
Polkadot API (PAPI) Integration PR

### PR DESCRIPTION
**Pull request by rossbulat**
This PR integrates Polkadot API into the staking dashboard, replacing and discontinuing Polkadot JS API.

Note: This refactor will have no effect on end users.

- [x] Initialisation
- [x] Constants
- [x] Runtime APIs
- [x] Storage Queries
- [x] Transaction Submission

### Commit Log
- [x] Initialise Polkadot API with Smoldot & Websocket Support, and retrieve chain spec.
- [x] Expose untyped API client.
- [x] Destroy pApi on disconnect.
- [x] Fetch dashboard constants via pApi.
- [x] Fetch the active era via pApi.
- [x] Fetch network metadata via pApi.
- [x] Move activeEra logic to a Subscription class.
- [x] Refactor BlockNumber Subscription to pApi class.
- [x] Add RXJS for typing and observable utility support.
- [x] Refactor NetworkMetrics Subscription to pApi class.
- [x] Refactor PoolsConfig Subscription to pApi class.
- [x] Refactor StakingMetrics Subscription to pApi class.
- [x] Fetch chain version metadata and add to chain specs.
- [x] Refactor fetching of validator entries via ValidatorEntries pApi class.
- [x] Refactor fetching of erasStakersOverview to ErasStakersOverview pApi class.
- [x] Refactor fetching of erasStakersPaged to ErasStakersPaged pApi class.
- [x] Refactor fetching of bonded pools via BondedPoolEntries pApi class.
- [x] Add dedicated directory for entries fetching
- [x] Refactor fetching of bonded accounts to a pApi Subscription class.
- [x] Use substrate-bindings to generate pool addresses.
- [x] Refactor Fast Unstake config to be a separate pApi Subscription class. 
- [x] Abstract Fast Unstake queue for an account into a separate Subscription class.
- [x] Abstract account balances to a new Subscription class.
- [x] Add fetching and formatting of one BondedPool.
- [x] Discontinue fetching pool member entries, no longer scalable.
- [x] Migrate pool multi queries to classes.
- [x] Move account proxies subscription to Account Proxies Subscription class. 
- [x] Migrate proxy queries to ProxyQuery class.
- [x] init ActivePoolAccount subscription & remove multi role logic.
- [x] Migrate IdentitiesController to pApi.
- [x] Migrate validator entries queries.
- [x] Add ParaSessionAccounts query to replace subscription.
- [x] Add PoolMemberBatchEvent and migrate pool member batches.
- [x] Migrate ValidatorEntries multi queries.
- [x] Migrate ClaimedRewards, BondedMulti queries for Payouts.
- [x] Migrate remaining Payouts queries.
- [x] Refactor tx hooks for pApi support.
- [x] Submit transactions via pApi.
- [x] Remove useBuildPayload hook.
- [x] Stop exposing api and peopleApi through API context.
- [x] Stop connecting to PJS API.
- [x] Uninstall PJS API dependencies.
- [x] Added custom signer for Ledger.
- [x] Ensure all queries and apis are fetched at best.
- [x] Added custom signer for Polkadot Vault.
- [x] Add custom signer for WalletConnect with amended submission flow.

### CODE REVIEW

**Comment by voliva** [comment]
> Since this effort started, the [untyped API](https://papi.how/unsafe), at our request, was added as a feature of the top-level PAPI API. This API makes it viable to be used for Dapps that do not assume the network or chain at development time (this was the main bottleneck we were communicating in the initial stages of Polkadot API integration). Since this was deployed, advisory changed from low-level integration to top level integration.

Hey @rossbulat 👋, I'd like to clarify a few points here, as this isn't entirely accurate.

First of all, it's called `unsafeApi`, instead of `untypedApi`. The `unsafeApi` actually supports fully-typed interactions, by passing the type of the descriptors into its generic, such as shown in [the docs](https://papi.how/unsafe/).

The main difference between the `typedApi` and the `unsafeApi` is that the `unsafeApi` skips the compatibility check of the actions you perform.

This has the upside that it doesn't require descriptors to be generated or bundled with the dApp. However, this comes at the cost of losing features like the `.isCompatible()` function, which is essential for handling runtime upgrades or interacting with different chains, which might be especially useful when supporting generic chains that may have different runtime versions.

Without these safety measures, you risk performing operations that could yield unexpected results or overlook additional parameters required for specific chains or runtime versions.

**The `typedApi` is perfectly capable of connecting to any chain, regardless of where the origin of those descriptors**. The descriptors are just a representation of the types shown in your IDE during development. At runtime, these types are compared to the actual runtime of the connected client. In the case the types don't match, then it guards you sending incorrect data in case the interaction has become incompatible (you can explore [different levels of compatibility](https://papi.how/typed#getcompatibilitylevel) in the documentation).

The `unsafeApi` was introduced as an escape hatch for scenarios where generating or bundling descriptors is challenging, for example, in environments like code sandboxes, UMD builds, and similar cases. This API was added in September (released in `polkadot-api@1.4.0`) following a [request from @tien](https://github.com/polkadot-api/polkadot-api/issues/705).

In your case, generating descriptors for the main chain you develop for would make the most sense. You could also generate descriptors for other chains with different interactions to maintain compatibility across chains, or even check in runtime which interaction is the current one.

Regarding low-level libraries versus the `polkadot-api`, our stance has consistently been that low-level libraries are intended for advanced use cases requiring access to chain primitives, such as when building a block explorer. And even then, in the developer console that we developed recently, we actually used a mix of the "top-level" `polkadot-api` and lower-level libraries because the top-level abstractions were still beneficial for some components. However, for dApps like the staking dashboard, governance manager, transfers UI, and similar applications, we've always recommended the top-level library.

Lastly, to prevent further misunderstandings, please, use the public channels we have available for all the community.

I hope you find this comment useful, and that it does clarify any misunderstanding you had 🙏

**Comment by rossbulat** [comment]
Yeah. I call it untyped API.

File: packages/app/src/model/Api/index.ts [hunk 1]
```diff
@@ -86,25 +101,29 @@ export class Api {
...
+      // NOTE: Unlike Polkadot JS API, Papi client does not have an asynchronous initialization
+      // stage that leads to `isReady`. If using papi client, we can immediately attempt to fetch
+      // the chainSpec via the client.∑
       this.initApiEvents();
 
       // Wait for api to be ready.
       await this.#api.isReady;
+
+      // Initialise PAPI API.
+      this.#pApi = this.#papiClient.getUnsafeApi();
```
Comment (voliva) [inline packages/app/src/model/Api/index.ts:116]: You should use `.getTypedApi()` instead

**Review by voliva — COMMENTED** [review]
> Please keep discussions of the API to external forums. This is for PR only. Reviews are welcome, anything else is not.

I'm finding this behaviour extremely offensive TBH. My comment was fully constructive, and it was referencing how this entire PR is based off on a misunderstanding you have of what you call the "untypedApi" (which is actually called the `unsafeAPI`) vs the `typedApi`. I've put valuable time trying to help and you just decided to just directly trash it. I didn't expect this from you.

I'm sorry if it came across as unsolicited advice, but given this is open-source and you were showing inaccuracies, I thought it would be helpful. If you don't want to take my advice, you are absolutely free to do so, as you are the owner of this project.

Anyway here's my review - The changes required shouldn't be that much, since the `typedApi` is forwards-compatible with the `unsafeApi`. Generate the descriptors of the chain with the interactions you use, and use those descriptors, changing only that single line.

The reason was explained throughly in the comment you deleted. As a tl;dr; the only difference between the `unsafeApi` and the `typedApi` is that the unsafeApi doesn't perform compatibility checks, because it doesn't have the descriptors in runtime. The descriptors are a representation of the types shown in your IDE, so that they can be compared with the types in the current runtime of the chain you're connected to. This means the `typedApi` can be used for any chain, regardless of where the descriptors were created from.

This is a review of this PR. You can either accept it or ignore it, but please do not delete it, as you did with the previous comment (which was also thought as a review of the PR)

**Comment by rossbulat** [comment]
> > Please keep discussions of the API to external forums. This is for PR only. Reviews are welcome, anything else is not.
> 
> I'm finding this behaviour extremely offensive TBH. My comment was fully constructive, and it was referencing how this entire PR is based off on a misunderstanding you have of what you call the "untypedApi" (which is actually called the `unsafeAPI`) vs the `typedApi`. I've put valuable time trying to help and you just decided to just directly trash it. I didn't expect this from you.
> 
> I'm sorry if it came across as unsolicited advice, but given this is open-source and you were showing inaccuracies, I thought it would be helpful. If you don't want to take my advice, you are absolutely free to do so, as you are the owner of this project.
> 
> Anyway here's my review - The changes required shouldn't be that much, since the `typedApi` is forwards-compatible with the `unsafeApi`. Generate the descriptors of the chain with the interactions you use, and use those descriptors, changing only that single line.
> 
> The reason was explained throughly in the comment you deleted. As a tl;dr; the only difference between the `unsafeApi` and the `typedApi` is that the unsafeApi doesn't perform compatibility checks, because it doesn't have the descriptors in runtime. The descriptors are a representation of the types shown in your IDE, so that they can be compared with the types in the current runtime of the chain you're connected to. This means the `typedApi` can be used for any chain, regardless of where the descriptors were created from.
> 
> This is a review of this PR. You can either accept it or ignore it, but please do not delete it, as you did with the previous comment (which was also thought as a review of the PR)

ChatGPT generated content is not useful, and what you posted is nothing we did not already know. You did not raise any issues at Sub0 after I gave you the opportunity, yet you still decide to troll this effort on Github. TLDR: It is our choice not to use descriptors at this time. This may change in the future, or it may not.

**Pull request merged**